We need to adjust the script that runs the tests to get them out of the
build directory not the original source directory and we need to be able
to specify an alternate re2c binary when running system tests.

This patch should be sent upstream.

--- re2c-0.14.2/run_tests.sh.in.orig	2015-10-13 06:22:20.787156312 -0700
+++ re2c-0.14.2/run_tests.sh.in	2015-10-13 06:33:42.229167381 -0700
@@ -14,7 +14,7 @@
 if uname|grep -q -i 'CYGWIN' && test -x "@WINBUILDDIR@/re2c.exe"; then
 re2c='@WINBUILDDIR@/re2c.exe'
 else
-re2c='@builddir@/re2c'
+re2c="${RE2C_BIN:-@builddir@/re2c}"
 fi
 if test ! -x "${re2c}"; then
   echo "Cannot find re2c executable (${re2c})."
@@ -33,7 +33,7 @@
 fi
 echo "Diffing: ${diff}"
 if test $# = 0; then
-tests=`for i in test lessons; do find @top_srcdir@/$i -name '*.re'; done|sort`
+tests=`for i in test lessons; do find ./$i -name '*.re'; done|sort`
 else
 tests="$@"
 fi;
